packages/server/graphql/mutations/resetMeetingToStage.ts

```diff
@@ -0,0 +1,134 @@
+  description: `Reset meeting to a previously completed stage`,
+  args: {
+    meetingId: {
+      type: GraphQLNonNull(GraphQLID)
+    },
+    stageId: {
+      type: GraphQLNonNull(GraphQLID)
+    }
+  },
+  resolve: async (_source, {meetingId, stageId}, {authToken, socketId: mutatorId, dataLoader}) => {
+    const r = await getRethink()
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+
+    // AUTH
+    const viewerId = getUserId(authToken)
+    const meeting = await dataLoader.get('newMeetings').load(meetingId)
+    if (!meeting) return standardError(new Error('Meeting not found'), {userId: viewerId})
+    const {
+      defaultFacilitatorUserId,
+      facilitatorUserId,
+      phases,
+      teamId,
+      meetingType,
+      meetingCount
+    } = meeting
+    if (viewerId !== facilitatorUserId) {
+      if (viewerId !== defaultFacilitatorUserId)
+        return standardError(new Error('Not meeting facilitator'), {userId: viewerId})
+      return standardError(new Error('Not meeting facilitator anymore'), {userId: viewerId})
+    }
+
+    // VALIDATION
+    const foundResponse = findStageById(phases, stageId)
+    if (!foundResponse)
+      return standardError(new Error('Meeting stage not found'), {userId: viewerId})
+    const {stage: resetToStage} = foundResponse
+    if (!resetToStage.isNavigableByFacilitator)
+      return standardError(new Error('Stage has not started'), {userId: viewerId})
+    if (!resetToStage.isComplete)
+      return standardError(new Error('Stage has not finished'), {userId: viewerId})
+    if (resetToStage.phaseType !== NewMeetingPhaseTypeEnum.group)
+      return standardError(new Error('Resetting to this stage type is not supported'), {
+        userId: viewerId
+      })
+    if (meeting.endedAt)
+      return standardError(new Error('The meeting has already ended'), {userId: viewerId})
+
+    // RESOLUTION
+    const createdPhases = await createNewMeetingPhases(
+      teamId,
+      meetingCount,
+      meetingType,
+      dataLoader
+    )
+    let shouldResetStage = false
+    let resetToPhaseIndex = -1
+    const newPhases = [] as GenericMeetingPhase[]
+    for (const [phaseIndex, phase] of phases.entries()) {
+      const {stages} = phase
+      if (!stages) continue
+      const newStages = [] as GenericMeetingStage[]
+      for (const [stageIndex, stage] of stages.entries()) {
+        if (stage.id === stageId) {
+          shouldResetStage = true
+          resetToPhaseIndex = phaseIndex
+        }
+        if (!shouldResetStage) {
+          newStages.push(stage)
+          continue
+        }
+        const resettedStage = createdPhases[phaseIndex]?.stages[stageIndex]
+        if (!resettedStage) continue
+        newStages.push(Object.assign(resettedStage, {id: stage.id}))
+      }
+      phase.stages = newStages
+      newPhases.push(phase)
+    }
+    primePhases(newPhases, resetToPhaseIndex)
+
+    const reflectionGroups = await dataLoader
```

Comment:
warning that the dataloader only makes 1 fetch per request. when requested again, it'll use the cached version.
since you need to mutate the `voterIds` field, you either need to not use the dataloader, or mutate the cached value in addition to mutating the record in the DB.

e.g.
```js
reflectionGroups.forEach((group) => group.voterIds = [])
```